src/dotnet/ToolPackage/ToolConfigurationAndExecutablePath.cs

```diff
@@ -5,17 +5,18 @@
 
 namespace Microsoft.DotNet.ToolPackage
 {
-    internal class ToolConfigurationAndExecutableDirectory
+    internal class ToolConfigurationAndExecutablePath
```

Comment:
Nit: I wonder if this type could simply be replaced with a value tuple in the future (note: not important for this PR).